diff --git a/spatial.cu b/spatial.cu
index 1a6a654..207fc3a 100644
--- a/spatial.cu
+++ b/spatial.cu
@@ -20,7 +20,7 @@ distCUDA2(const torch::Tensor& points)
   auto float_opts = points.options().dtype(torch::kFloat32);
   torch::Tensor means = torch::full({P}, 0.0, float_opts);
   
-  SimpleKNN::knn(P, (float3*)points.contiguous().data<float>(), means.contiguous().data<float>());
+  SimpleKNN::knn(P, (float3*)points.contiguous().data_ptr<float>(), means.contiguous().data_ptr<float>());
 
   return means;
 }
\ No newline at end of file
